File: 35-search-insert-position/search-insert-position.cpp
// class Solution {
// public:
//     int searchInsert(vector<int>& nums, int target) {
//         int lowest=nums.size();
//         int low=0;
//         int high=nums.size()-1;
//         while(low<=high){
//             int mid=(low+high)/2;
//             if(nums[mid]>=target){
//                 lowest=mid;
//                 high=mid-1;
//             }
//             else{
//                 low=mid+1;
//             }
//         }
//         return lowest;
//     }
// };




class Solution {
public:
    int searchInsert(vector<int>& arr, int target) {
        int lowest=arr.size();
        int low=0;
        int high=arr.size()-1;
        while(low<=high){
            int mid=(low+high)/2;
            if(arr[mid]>=target){
                lowest=mid;
                high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        return lowest;
    }
};

File: 35-search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # edge cases
        if(target<nums[0]):
            return 0
        if(target>nums[-1]):
            return len(nums)
        low=0
        high=len(nums)-1
        res=-1
        while(low<=high):
            mid=(low+high)//2
            if(nums[mid]==target):
                return mid
            elif(nums[mid]>=target):
                res=mid
                high=mid-1
            else:
                low=mid+1
        return res